apache2/waf_logging/waf_format.proto

```diff
@@ -19,19 +19,15 @@ message Properties {
   optional string ruleSetVersion = 6;
   optional string ruleId = 7;
   optional string message = 8;
-
-  enum ActionType {
-    Detected = 0;
```

Comment:
Just a thought, could we create a separate PR for the clean up work here? Would be easier to parse those from a historical perspective as well. The problem with mixing all the commits is that it conflates the PR if we do a Merge and Rebase at the very end.

Reply:
Well some clean-up is needed here, like removal of unused parameters from helpers because if previously a `ngx_http_request_t->pool` has been passed in, now there is simply no `ngx_http_request_t` in some contexts where it is being called.

But all the dead code removal can go in a separate PR. Let me split the changes and update this one.